lib/inference/open_ai_image_generator.rb

```diff
@@ -40,6 +40,11 @@ def self.create_uploads!(
             output_format: output_format,
           )
 
+        raise api_responses[0] if api_responses.all? { |resp| resp.is_a?(StandardError) }
+
+        api_responses =
+          api_responses.filter_map { |response| response.is_a?(StandardError) ? nil : response }
```

Comment:
```suggestion
        api_responses =
          api_responses.filter { |response| !response.is_a?(StandardError) }
```
Do you need those `nil` leftover from StandardError?

Reply:
agree this is simpler ... filter_map removed them, but no point mapping